scss/_utilities/_mediaqueries.scss

```diff
@@ -0,0 +1,43 @@
+/* 
+* Media Query Mixin  
+* Containes logic to easily remove media queries for
+* desktop only or mobile only projects
+* Examples:
+* 1. @include mq( 300px ) { } (min-width)
+* 2. @include mq( $max: 300px ) { } (max-width)
+* 3. @include mq( 400px, 800px ) { } (both min + max-width)
+*/
+$media-queries: true;
+$desktop-styles: true;
+
+@mixin mq($min: null, $max: null) {
+	@if $min == null and $max == null {
+		@error 'Please provide a value for $min and/or $max';
+	}
+
+	@if $media-queries {
+		@if $min != null and $max != null {
+			@media (min-width: $min) and (max-width: $max) {
+				@content;
+			}
+		}
+		@else if $min != null {
+			@media (min-width: $min) {
+				@content;
+			}
+		}
+		@else {
+			@media (max-width: $max) {
+				@content;
+			}
+		}
+	}
+	@else if $desktop-styles and $max == null {
+		/* Content printed with no media query for desktop project */
+		@content;
+	}
+	@else if $desktop-styles == false and $min == null {
```

Comment:
Yes the idea behind the $desktop-styles variable is to only be used if $media-queries is set to false. This way the code knows if it should print out styles based on styles targeted to large screens vs small screens.

As for a third parameter I'm having a hard time imagining what that would be. I imagine the goal of the third parameter is to ensure the user is getting all the styles they need. When writing the code initially I was thinking the lowest size screen styles would be "mobile" while highest would be "desktop" but I can see a point where that might not be the case. While pondering this question it did make me think of an idea (sorry if this is what you were saying and I totally misunderstood) maybe we should remove the $desktop-style variable and replace it with a variable called $app-size (or something like that) that takes a pixel size that defines the screen size the user wishes to support without media queries.

For example if I was making an iPad app I would pass 1024px to the variable (iPad landscape). Then when styles are evaluated through the mixin we could check if the min/max width fits the size to determine if it should be printed or not.

Please let me know if this didn't address your question but I do really like this idea and if others are on board with it I will add it to the mixin. Thanks for reviewing!